### ISSUE TITLE
Different behavior on set user traits between Android and iOS sdk

### ISSUE THREAD

**Issue by gumil** [Survicate/survicate-android-sdk#9]
We noticed there's a difference between Android and iOS sdks on how it handles custom traits. In iOS when you set a custom trait it can be saved but in Android it needs to be set all at once. Is this the expected operation of this method?

In Android:
```kotlin
List<UserTrait> traits = new ArrayList<>();
traits.add(new UserTrait.UserId("someUserId"));
traits.add(new UserTrait.FirstName("John"));
traits.add(new UserTrait("eyes", "blue"));
Survicate.setUserTraits(traits);
```
You need to set the user traits all at once.

In iOS/Swift setting the user traits in different places is possible and gets sent as a response.

**Comment by j-brzozowski** [comment]
Have you tried this:
`Survicate.setUserTrait(new UserTrait.UserId("someOtherUserId"));` ?

**Comment by gumil** [comment]
Yes looking at the decompiled code that method just creates a new list

**Comment by michal-survicate** [comment]
I tested it and it seems that in android SDK you need to meet the conditions in order, starting down and going up. In iOS order doesn't matter.

**Comment by michal-survicate** [comment]
Anyways - we'll try to fix that issue in next release :)

**Comment by gumil** [comment]
I like the behavior of Android better. It makes it not rely so much on the Survicate singleton state

**Comment by j-brzozowski** [comment]
It is fixed in 1.1.2
